### ISSUE TITLE
Translation Not Working in Checkout 2.2.1

### ISSUE THREAD

**Issue by eCommerceGorilla** [magento/magento2#12350]
Created a translation file en_US.cs in a theme in app/design/frontend/Company/theme/i18n/
Changes made to the translation csv are not being applied to checkout

For example added the following to the translation csv:
"Apply Discount Code","Have a Promotional Code?"
"Apply Discount","Apply Promotion"

Cleared translation cache and reloaded and no change to the checkout. However the translation is applied to the shopping cart.

To be clear, the translation is working on the shopping cart page and other places throughout the site just not in the onepage checkout.

Have tested this on both blank and luma theme as well and the issue persists.

Magento 2.2.1
PHP 7.0.25

**Comment by matin73** [comment]
I think same issue as mine here https://github.com/magento/magento2/issues/12348

**Comment by cytracon** [comment]
Had the same problem. That
https://mage2.pro/t/topic/4524/2
fixed it for me.

**Comment by magento-engcom-team** [comment]
Duplicate #12348

**Comment by eCommerceGorilla** [comment]
@magento-engcom-team  not a duplicate of at all of the post you mentioned. In fact besides just being about translation they share no similarities.

@cytracon Thank for the suggestion, I reviewed and tested but wasn't related to the issue I'm having unfortunately.

What I have been able to determine is that the js-translation.json files are not being generated. This appears to be affecting the checkout.

It appears it might be related to https://github.com/magento/magento2/issues/7525#issuecomment-316690100

Which is another topic that was closed back in 2016 without a proper resolution as users are still experiencing the issue.

**Comment by palo** [comment]
Also struggling with this issue...  in 2.2.1

**Comment by eCommerceGorilla** [comment]
@palo if you come up with a solution please share it here. I've been working on this but I keep hitting a dead end.

**Comment by eCommerceGorilla** [comment]
@magento-engcom-team re-open this was entirely unrelated to #12348 please read the entire issue before closing next time.

**Comment by palo** [comment]
@eCommerceGorilla Magento 2.2.1 "quick strategy" seemed suspicious from the first time i saw it.

This solved **almost** all js translation problems:
`bin/magento  setup:static-content:deploy xx_XX --theme="vendor/theme" -s standard`

@magento-engcom-team should fix quick strategy deployment or update documentation including this _feature_ in http://devdocs.magento.com/guides/v2.2/config-guide/cli/config-cli-subcommands-static-deploy-strategies.html

**Comment by eCommerceGorilla** [comment]
I wanted to update this with what I have found.

First, in 2.2.1 it appears that editing a language csv such as: 
`app/design/frontend/Vendor/theme/i18n/en_US.csv`

Has no effect on js-translate.json files in regards to modules such as is the case with the checkout. However, this approach seems to update almost any other string inside the website.

Instead, what I had to do was the following:
1) create a collection of phrases with `php bin/magento i18n:collect-phrases -o en_US.csv -m`
2) Create a language pack `http://devdocs.magento.com/guides/v2.0/frontend-dev-guide/translations/xlate.html#m2devgde-xlate-languagepack`
3) Make my changes to the language pack and upload then run `php bin/magento setup:static-content:deploy en_US` (Note if you are running in developer mode and previously deploy static content you will need to run `bin/magento setup:static-content:deploy en_US -f`  otherwise it will use the deployed translation even in developer mode. The `-f` just forces the deployment)

This works for me. However I find it strange that simply updating the theme language strings doesn't cascade into the checkout strings.
